std/experimental/allocator/building_blocks/kernighan_ritchie.d

```diff
@@ -117,13 +117,21 @@ struct KRRegion(ParentAllocator = NullAllocator)
         {
             assert(right);
             auto p = payload;
-            return p.ptr + p.length == right;
+            return p.ptr < right && right - (p.ptr + p.length) < Node.sizeof;
```

Comment:
All other things being equal, prefer '+' to '-' so maintainer doesn't need to think about positive and negative results (in this case, for example, a negative result on the left will be converted to unsigned before comparison).

```d
return p.ptr < right && right < p.ptr + p.length + Node.sizeof;
```